api/mock_server/app.py

```diff
@@ -36,24 +36,37 @@
     scenarios=scenarios
 )
 
+app = connexion.App(__name__, specification_dir="../",
+                    options={"swagger_ui": False})
+
 
 def login_post():
     token = state.login()
     print(f"got token: {token}")
-    if token is not None:
-        return ({"X-Auth-Token": token, "authentication_url": "http://todo"}, 200)
-
-    return ({"message": "Invalid username/password",
-             "code": "invalid_credentials"}, 401)
+    host = connexion.request.host
+    return (
+        {
+            "X-Auth-Token": token,
+            "authentication_url": f"http://{host}/authorize-login?token={token}"
+        },
+        200
+    )
+
+@app.route('/authorize-login')
+def authorize_login_post():
+    token = connexion.request.args.get("token")
+    fail = connexion.request.args.get("fail") is not None
+    state.authorize_login(token, fail)
+    return "", 204
```

Comment:
What is the point of `fail` here?

Reply:
I suppose to allow making an invalid token for testing purposes. Maybe it should be made easier to do after having initially logged in successfully?